web-app/packages/lib/src/modules/project/views/VersionDetailView.vue

```diff
@@ -181,31 +181,25 @@ export default Vue.extend({
     $route: 'getVersion'
   },
   methods: {
-    getVersion() {
-      if (!this.project.versions) {
-        ProjectApi.getProjectVersion(
-          this.project.namespace,
-          this.project.name,
-          this.version_id
-        )
-          // TODO: types
-          .then((resp: any) => {
-            if (resp.data.length) {
-              const version = resp.data[0]
-              version.changes = resp.data[0].changes
-              this.version = version
-            }
-          })
-          .catch((err) => {
-            const msg = err.response
-              ? err.response.data?.detail
-              : 'Failed to fetch project version'
-            this.$store.dispatch('notificationModule/error', { text: msg })
-          })
-      } else {
-        this.version = this.project.versions.find(
+    async getVersion() {
+      try {
+        const currentVersion = this.versions?.find(
           (v) => v.name === this.version_id
         )
+        if (!currentVersion) {
+          const response = await ProjectApi.getProjectVersion(
+            this.project.id,
+            this.version_id
+          )
+          this.version = response.data
+          return
+        }
+        this.version = currentVersion
```

Comment:
I would do this other way around. If found version in version, this.version = currentVersion and return (even before try clause) else call api, etc

Reply:
Refactor in commit https://github.com/MerginMaps/mergin/pull/74/commits/b27ba5e04dd739a40521fb55f437dd8834bfcbeb